### PR TITLE
Tmp dir overrides

### PR DESCRIPTION
**Pull request by racergoodwin**
This request adds more options to the tempdir overrides, allowing for different start criteria of multiple tempdirs.

### CODE REVIEW

**Comment by altendky** [comment]
Could you do a catch up merge and I'll take another look?  There are type hints now so if you have to wait for me to approve again after the update, you can give `tox -e check-hints-py39` (or whatever python version you have) a try locally for feedback on that.

One point I'm not sure on, like not sure which I actually prefer, is whether the override should be allowed to be incompletely specified.  I understand reducing repetition, but does it really make sense to be able to override just the phase major, for example?

**Comment by racergoodwin** [comment]
Sure thing.
Yeah, it could cause confusion if it isn't necessary to specify major and minor phases. Will change that behaviour.

**Comment by altendky** [comment]
I also started adding brief lines in the changelog with links to the PRs.  Could you please add one for this while you're at it?

File: src/plotman/manager.py
Comment (altendky) [inline src/plotman/manager.py:85]: I don't have a formal argument for this, but I think I would rather keep this more in the form it was before.

```suggestion
    if len([p for p in phases if p < milestone]) >= stagger_phase_limit
        return False

    if len(phases) >= max_plots:
```

File: src/plotman/manager.py
Comment (altendky) [inline src/plotman/manager.py:88]:
```suggestion

    return True
```

File: src/plotman/manager.py
Comment (altendky) [inline src/plotman/manager.py:82]:
```suggestion
    milestone = job.Phase(major,minor)
```

File: src/plotman/manager.py
Comment (altendky) [inline src/plotman/manager.py:100]: The stagger seconds was added not too long ago.  I personally plan on going the other way and considering how to indicate the status of all restrictions at all times.  Was there a particular need to remove this?

```suggestion
        wait_reason = 'global max jobs reached (%d) - (%ds/%ds)' % (sched_cfg.global_max_jobs, youngest_job_age, global_stagger)
```

Side note, and no need to change it now, but this PR probably should have been at least four.  Offhand, the new per-tmp phase stagger, the addition of global, removal of the time-stagger indication here, and moving of the overrides to scheduling.  Just a thought for next time.  Small PRs are quicker to verify and merge (and undo if needed).

File: src/plotman/manager.py
Comment (altendky) [inline src/plotman/manager.py:109]: Is this actually limited to indicating phase-related restrictions?  I quickly checked `phases_permit_new_job()` and it looks like it also checks the process limit.

https://github.com/ericaltendorf/plotman/blob/d52f51600736dbb2a1635ab80eede606949f1903/src/plotman/manager.py#L47-L74

File: src/plotman/manager.py
Comment (altendky) [inline src/plotman/manager.py:98]: If there's something to clarify wouldn't it be that this is for the time-stagger not the phase-stagger?  Though the reporting of seconds adjacent to the `stagger` comment it seems to lead that way.

**Review by altendky — CHANGES_REQUESTED** [review]
Since this is a backwards incompatible configuration change we should bump `plotman.configuration.expected_major_version` to `2`.  We'll need to add some commentary to the [wiki configuration page](https://github.com/ericaltendorf/plotman/wiki/Configuration#versions) as well.

File: src/plotman/manager.py
Comment (racergoodwin) [reply src/plotman/manager.py:85]: No worries. I will shift it back.
To my eye, it looked neater and was easier to understand with less repetition

File: src/plotman/manager.py
Comment (racergoodwin) [reply src/plotman/manager.py:98]: It seems that the only relevant need for the seconds to be displayed is for the global stagger m setting.
To me, it looks poor, having a clock that goes passed it's target and keeps ticking when it is no longer relevant.
Maybe keep showing youngest_job_age and remove global_stagger, once global_stagger has passed?
I had thought about adding an option of putting individual running time counters (hh:mm) in front of each running plot so people would know their overall duration, but as a different PR.

**Comment by racergoodwin** [comment]
Thanks for looking at this. I will get onto them all this evening.

File: src/plotman/manager.py
Comment (altendky) [reply src/plotman/manager.py:101]: It lets you know how far past the time stagger you have gone based on some other restriction.  Going beyond the time stagger isn't inherently bad. 
 I specifically set my time stagger a bit less than even spacing would be based on my total jobs and job duration.  Anyways, to satisfy "everyone" I expect I'll need to make most of the pieces optional.  There's a PR adding a couple lines and another adding a couple columns and...  `:]`  And I don't super want to do that in curses when I've already started a switch to prompt_toolkit.

**Comment by altendky** [comment]
Just a quick note, the failing macOS build looked like a general CI setup failure of some sort so I reran all the jobs.  Looks like they pass now.

**Comment by altendky** [comment]
I unresolved a couple "conversations" that were marked resolved without related code changes or replies.  Neither seem to be related to the core of this PR so it would be good to not hold the whole thing up for too long on them.  `:]`

**Comment by racergoodwin** [comment]
Fair enough. I thought I'd clicked on all the commit suggestions?

File: src/plotman/manager.py
Comment (racergoodwin) [reply src/plotman/manager.py]: all wait_reason defines returned to original

File: CHANGELOG.md
Comment (altendky) [inline CHANGELOG.md:14]: The "overrides" were already there, right?  The phase limit control was added.

```suggestion
- tmp directory overrides moved to `scheduling:` `tmp_overrides:`.
  ([#758](https://github.com/ericaltendorf/plotman/pull/758))
- Per tmp directory phase limit control added to `scheduling:` `tmp_overrides:`.
  ([#758](https://github.com/ericaltendorf/plotman/pull/758))
```

**Review by altendky — COMMENTED** [review]
Thanks again for the work here.  I think what's here is good and I can handle the somewhat separate layers of configuration structure changes that we talked about.  Would you be comfortable with me adding that into this PR?

**Pull request merged**
